swift-sdk/Internal/RequestCreator.swift

```diff
@@ -509,20 +505,16 @@ struct RequestCreator {
             ITBError(Self.authMissingMessage)
             return .failure(IterableError.general(description: Self.authMissingMessage))
         }
+
         
-        var body: [AnyHashable: Any] = [JsonKey.platform: JsonValue.iOS,
-                                        JsonKey.systemVersion: UIDevice.current.systemVersion,
-                                        JsonKey.Embedded.sdkVersion: IterableAPI.sdkVersion]
-        
-        if let packageName = Bundle.main.appPackageName {
-            body[JsonKey.Embedded.packageName] = packageName
-        }
+        var body = [AnyHashable: Any]()
         
         setCurrentUser(inDict: &body)
         
-        // TODO: find/create proper key for the value of the embedded message ID
+        body.setValue(for: JsonKey.messageId, value: message.metadata.messageId)
+        body.setValue(for: JsonKey.deviceInfo, value: deviceMetadata.asDictionary())
         
-        return .success(.post(createPostRequest(path: Const.Path.embeddedMessageImpression, body: body as! [String: String])))
+        return .success(.post(createPostRequest(path: Const.Path.embeddedMessageImpression, body: body)))
```

Comment:
Seems like we do not have an impression request. I think we should remove if we are not using it.

Reply:
I can cut a separate ticket for this.